modules/biobank/php/optionsendpoint.class.inc

```diff
@@ -0,0 +1,166 @@
+<?php declare(strict_types=1);
+/**
+ * This implements the Options endpoint class for the Biobank Module.
+ *
+ * PHP Version 7.2
+ *
+ * @category   API
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.text GPLv3
+ * @link       http://www.github.com/aces/Loris
+ */
+namespace LORIS\biobank;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * A class for handling the /biobank/options/ endpoint.
+ *
+ * PHP Version 7.2
+ *
+ * @category   API
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.text GPLv3
+ * @link       http://www.github.com/aces/Loris/
+ */
+class OptionsEndpoint implements RequestHandlerInterface
+{
+    /**
+     * Returns true if user has access to this endpoint.
+     *
+     * @param \User $user The user whose access is being checked.
+     *
+     * @return bool
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return true;
+    }
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return ['GET'];
```

Comment:
```suggestion
        return ['OPTIONS'];
```
? Unless I am misunderstanding

Reply:
So when someone sends a request with method 'OPTIONS', this class will return what methods options are available for request. In this case: `GET`.

This is the way @xlecours has it set up for his APIs so perhaps he can provided more clarity.

Reply:
I guess we should add OPTIONS to the allowed methods where it is implemented. I'll do it on https://github.com/aces/Loris/pull/4244

Reply:
done.